src/main/java/mzaba/avroprotocol/kafka/producer/SpringAvroProducer.java

```diff
@@ -0,0 +1,34 @@
+package mzaba.avroprotocol.kafka.producer;
+
+import lombok.RequiredArgsConstructor;
+import mzaba.avroprotocol.schema.Book;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.stereotype.Service;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+@Service
+@RequiredArgsConstructor
+public class SpringAvroProducer {
+
+    @Value("${avro.topic.name}")
+    String topicName;
+    private KafkaTemplate<String, Book> kafkaTemplate;
```

Comment:
`@RequiredArgsConstructor` generates constructor with **final** fields so **final** keyword should be added to `kafkaTemplate` to let Spring use constructor instead of autowiring with reflection